core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooQueueLock.java

```diff
@@ -42,6 +42,19 @@
   private String path;
   private boolean ephemeral;
 
+  public static class FateLockPath {
+    private final String path;
+
+    public FateLockPath(String path) {
+      this.path = path;
+    }
+
+    @Override
```

Comment:
Would getPath() improve readability vs. toString()?

Reply:
I did have a few different versions of this. One with getPath and I even had a method in `ZooLock` and `ZooQueueLock` to get the lock path at one point. They all seemed readable to me so I stuck with what seemed to be the simplest approach. If you think it adds good value to this then I can definitely add that back in. 

To be clear, are you suggesting renaming toString() to getPath() or something else?

Reply:
Probably depends when you ask me - on the first I saw toString() I wanted to confirm that it was just the path - getPath() would have been clearer to me and followed more of a getter / setter pattern (well getter since the class is immutable)  If there were additional convenience methods added, then it probably would make more sense.  Adding additional convenience functions if there are common code access patterns would be a stronger encapsulation of the (X)Lock concept as a object vs just a wrapper around a String.  But I think a lot of this is also personal preference - that's why I am trying to frame these questions to see if others had an opinion.

Reply:
I think there are 3 identical threads of conversation about `toString()` vs. `getPath()`, so we don't keep responding to them separately, I'm just going to mark two of them as "Resolved", and if there's more to discuss, then it can be discussed on the one that is left "Unresolved".